### PR TITLE
Fix display of lint messages without columns

### LINKED ISSUE #4336: Linting doesn't work if no column is provided by linting tool
### Steps 
1. Set the linting tool to something that uses the [flycheck](https://www.flycheck.org/) format (e.g. `set-option window lintcmd "mix credo list --config-file=.credo.exs --format=flycheck"` for an elixir project with credo)
1. Provoke a linting error that produces a linting error with no specified column (e.g. `foo.ex:42: error: use rust instead`)
1. Lint the buffer

### Outcome 
The Linting output is ignored completely.

### Expected 
The error should be handled, despite the absence of a column.

### PR DESCRIPTION
**Pull request by jtrees**
Fixes #4336.

### CODE REVIEW

File: rc/tools/lint.kak [hunk 1]
```diff
@@ -89,7 +89,16 @@ define-command \
                     '
                         BEGIN { OFS=":"; FS=":" }
 
-                        /:[1-9][0-9]*:[1-9][0-9]*:/ {
+                        # without column
+                        /^[^:]+:[1-9][0-9]*: / {
+                            $1 = ENVIRON["kak_bufname"]
+                            $2 += line_offset
+                            $2 = $2 FS "x";
```
Comment (mawww) [inline rc/tools/lint.kak:96]: can we get away with an empty string instead of a magic 'x' here ?

File: rc/tools/lint.kak [hunk 2]
```diff
@@ -89,7 +89,16 @@ define-command \
...
+                            print $0
+                        }
+
+                        # with column
+                        /^[^:]+:[1-9][0-9]*:[1-9][0-9]*: / {
```
Comment (mawww) [inline rc/tools/lint.kak:101]: could not capture groups do the trick to mostly share the code between with/without column ?

**Comment by mawww** [comment]
@lenormf Can you expand on the issue you see with this ?

**Comment by lenormf** [comment]
I’m not convinced this error format is ubiquitous enough to warrant the complexity in the plugin.

Plus our stance has been that if the output format is different than the “emacs” one, the linter should be wrapped in a shell script or post-processed in `lintcmd` directly.

The first option is not convenient because it makes users create random local scripts, the second requires knowledge of how the plugin runs `lintcmd` (`run() { … } && run`) — so I’d rather see #2336 merged so we can avoid handling corner cases like the `flycheck` format in Awk.

**Comment by jtrees** [comment]
> Plus our stance has been that if the output format is different than the “emacs” one, the linter should be wrapped in a shell script or post-processed in `lintcmd` directly.

Is this really the case though? I tried to find a specification for the format but couldn't find one. In fact, I think I found a snippet somewhere in the docs of the "emacs one" that implied the format without a column is also valid. I am simply recalling from memory though, don't have a source ready.

If it's not, I might simply try to get this changed upstream (in [credo](https://github.com/rrrene/credo)) instead.

File: rc/tools/lint.kak (hunk 1 above)
Comment (jtrees) [reply rc/tools/lint.kak]: Yes, good idea: b83b576028685b6d160f1fb2e7ca2aae5c0f5a6f

File: rc/tools/lint.kak (hunk 2 above)
Comment (jtrees) [reply rc/tools/lint.kak]: I'm not too familiar with awk and it looked like working with capture groups is either non-trivial or non-portable.

Here's my attempt at making the code more DRY: b83b576028685b6d160f1fb2e7ca2aae5c0f5a6f

Does this work for you?

File: rc/tools/lint.kak [hunk 3]
```diff
@@ -89,12 +89,14 @@ define-command \
                     '
                         BEGIN { OFS=":"; FS=":" }
 
-                        /:[1-9][0-9]*:[1-9][0-9]*:/ {
+                        /^[^:]+:([1-9][0-9]*:){1,2} / {
                             $1 = ENVIRON["kak_bufname"]
-                            if ( $2 == 1 ) {
+                            $2 += line_offset
+                            if ( $3 ~ /^ / ) {
+                                $2 = $2 FS
```
Comment (mawww) [inline rc/tools/lint.kak:96]: why do we need to add that `:` to $2 ? cant we use this logic (`$3 ~ /^ /`) in the subsequence awk script to detect if we had a column ?

File: rc/tools/lint.kak [hunk 4]
```diff
@@ -89,12 +89,14 @@ define-command \
...
+                            } else if ( $2 == 1 + line_offset ) {
```
Comment (mawww) [inline rc/tools/lint.kak:97]: I dont like adding line offset then comparing with line_offset + 1, it seems convoluted, surely we can just store that information in a boolean like variable if we need to check it before modifying $2.

**Review by mawww — COMMENTED** [review]
A few additional changes i'd like.

**Comment by jtrees** [comment]
Sorry for abandoning this. I have moved on from Kakoune (to Helix) and lost interest.

**Pull request closed without merging**